RadixWallet/Features/GatewaySettingsFeature/Components/GatewayList/GatewayList+View.swift

```diff
@@ -14,22 +12,32 @@ extension GatewayList {
 		}
 
 		public var body: some SwiftUI.View {
-			LazyVStack(spacing: .zero) {
-				ForEachStore(
-					store.scope(
-						state: \.gateways,
-						action: { .child(.gateway(id: $0, action: $1)) }
-					),
-					content: {
-						GatewayRow.View(store: $0)
-						Separator()
-							.padding(.horizontal, .medium3)
-					}
-				)
-			}
-			.buttonStyle(.tappableRowStyle)
-			.onAppear {
-				store.send(.view(.appeared))
+			WithViewStore(store, observe: { $0 }) { viewStore in
+				LazyVStack(spacing: .zero) {
+					ForEachStore(
+						store.scope(
+							state: \.gateways,
+							action: { .child(.gateway(id: $0, action: $1)) }
+						),
+						content: { store in
+							VStack(spacing: .zero) {
+								GatewayRow.View(store: store)
+
+								if store.gateway.id != viewStore.gateways.last?.gateway.id {
+									Separator()
+										.padding(.horizontal, .medium3)
+								} else {
+									Separator()
+								}
+							}
+							.background(Color.app.white)
```

Comment:
You can probably get rid of `Color` here, even if you needed it while code completing.